### PR TITLE
UrerF staging bug fixed

### PR DESCRIPTION
**Pull request by meghana-madhyastha**
@falkben @MrAE I believe this fixes the bug

### CODE REVIEW

File: packedForest/src/forestTypes/unsupervisedForests/stratifiedInNodeClassIndicesUnsupervised.h [hunk 1]
```diff
@@ -66,30 +63,45 @@ namespace fp{
...
 				for(int n=0; n<numUnusedObs; ++n){
 					outSamples[fpSingleton::getSingleton().returnLabel(potentialSamples[randomObsID])].push_back(potentialSamples[n]);
-				
 					outSamps.push_back(potentialSamples[n]);
 					}
 				
 			}
 
+			inline void createInAndOutSetsBagging(const int &numObs, float bagging){
+                                //TODO: We might want to refactor this when we move this over to the binned version.
```
Comment (falkben) [inline packedForest/src/forestTypes/unsupervisedForests/stratifiedInNodeClassIndicesUnsupervised.h:72]: somehow we seem to have a mix of tabs and spaces. i don't care which we use -- the C++ was mostly tabs as James wrote it, but we should stick to one or the other -- otherwise things are formatted weird

File: packedForest/src/forestTypes/unsupervisedForests/stratifiedInNodeClassIndicesUnsupervised.h [hunk 2]
```diff
@@ -164,7 +176,7 @@ namespace fp{
 			}
 
 			inline void initializeBinnedSamples(){
-				if(useBin()){
+				/*if(useBin()){
```
Comment (falkben) [inline packedForest/src/forestTypes/unsupervisedForests/stratifiedInNodeClassIndicesUnsupervised.h:179]: styling -- can we avoid block comments in general?  it makes it difficult for git history to show changes on a particular line

in this case -- perhaps just delete the code block?

File: packedForest/src/forestTypes/unsupervisedForests/stratifiedInNodeClassIndicesUnsupervised.h [hunk 3]
```diff
@@ -18,22 +19,18 @@ namespace fp{
...
-				createInAndOutSets(numObservationsInDataSet);
-				for(auto inSampsClass : inSamples){
-					inSampleSize += inSampsClass.size();
-				}
-
-				for(auto outSamps1 : outSamples){
-					outSampleSize += outSamps1.size();
-				}
+			stratifiedInNodeClassIndicesUnsupervised(const int &numObservationsInDataSet): inSampleSize(0), outSampleSize(0){
+				impurity = 10; //initialize to an arbitrary non zero value
```
Comment (falkben) [inline packedForest/src/forestTypes/unsupervisedForests/stratifiedInNodeClassIndicesUnsupervised.h:30]: thanks for the comment.  i was curious about why 10.  

My understanding is this needs to be > 0 so that we process the node (unless impurity really is 0).  can you update comment with that info?

**Review by falkben — CHANGES_REQUESTED** [review]
thanks so much for this.

just a few small changes listed inline.

Additionally:

We really need a "system" test for urerf to ensure it's working.  i can hopefully submit a PR to this branch with a test in python and a class that uses the same defaults as `R`'s UrerF.

Can we seed and get repeatable results as in the other forests?

**Comment by falkben** [comment]
One more minor thing: when I run urerf, I get a lot of printed "Not applicable for unsupervised random forests output"

**Comment by meghana-madhyastha** [comment]
> One more minor thing: when I run urerf, I get a lot of printed "Not applicable for unsupervised random forests output"

So, I remember in my previous PR we decided not to remove the functions that are meant for supervised RerF (eg returnMaxClass()) and also decided not to just return 0. So a suggestion was to print an error message.

**Comment by falkben** [comment]
> > One more minor thing: when I run urerf, I get a lot of printed "Not applicable for unsupervised random forests output"
> 
> So, I remember in my previous PR we decided not to remove the functions that are meant for supervised RerF (eg returnMaxClass()) and also decided not to just return 0. So a suggestion was to print an error message.

If these are just for supervised, why are they getting called?  If it was my suggestion we print the messages -- I guess I take it back.  Sorry :/

**Comment by meghana-madhyastha** [comment]
> > > One more minor thing: when I run urerf, I get a lot of printed "Not applicable for unsupervised random forests output"
> > 
> > 
> > So, I remember in my previous PR we decided not to remove the functions that are meant for supervised RerF (eg returnMaxClass()) and also decided not to just return 0. So a suggestion was to print an error message.
> 
> If these are just for supervised, why are they getting called? If it was my suggestion we print the messages -- I guess I take it back. Sorry :/

I think removing them broke some things. But let me try again

**Comment by meghana-madhyastha** [comment]
> > > One more minor thing: when I run urerf, I get a lot of printed "Not applicable for unsupervised random forests output"
> > 
> > 
> > So, I remember in my previous PR we decided not to remove the functions that are meant for supervised RerF (eg returnMaxClass()) and also decided not to just return 0. So a suggestion was to print an error message.
> 
> If these are just for supervised, why are they getting called? If it was my suggestion we print the messages -- I guess I take it back. Sorry :/

Anyway since I removed some of the inheritances I was able to remove these functions. I just updated the following branch with the fix: meghana_urerfbranchfix_test

**Comment by falkben** [comment]
Continued in #308 @MrAE
